controller/lib/src/external_api/generic/model.rs

```diff
@@ -1,10 +1,21 @@
 use serde::{Deserialize, Serialize};
-pub enum FilterError {
-    OutOfRange,
-}
 
 #[derive(Deserialize, Serialize)]
 pub struct Pagination {
     pub limit: u32,
     pub offset: u32,
 }
+
+#[derive(Deserialize, Serialize)]
+pub struct APIResponse {
+    pub data: Option<String>,
+    pub metadata: APIResponseMetadata,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct APIResponseMetadata {
+    pub error: Option<String>,
+    pub message: Option<String>,
+    pub count: Option<u32>,
+    pub status: u16,
```

Comment:
I'm not sure why you would add a status here ? The status is in the HTTP headers already, no need to duplicate the information here.

Reply:
I was just adding this to metadata so this object can have all information needed (as well as the status) but I can remove it in fact, it won't be so useful for us.